rpc/src/v1/helpers/dispatch.rs

```diff
@@ -371,7 +381,13 @@ impl Dispatcher for LightDispatcher {
 		match (request_nonce, force_nonce) {
 			(_, false) | (Some(_), true) => Box::new(gas_price),
 			(None, true) => {
-				let next_nonce = self.next_nonce(from);
+				let next_nonce = {
+					if let Ok(nonce) = self.cached_next_nonce(&from) {
```

Comment:
is there any chance to incapsulate this particular logic?

Reply:
This will go away too, but I just realized that this might perform two network requests in case the nonce is not cached and I need to changer this, thanks